src/main/java/frc/robot/commands/ShooterEject.java

```diff
@@ -0,0 +1,68 @@
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.Constants.AimModes;
+import frc.robot.subsystems.CDSSubsystem;
+import frc.robot.subsystems.ShooterSubsystem;
+
+public class ShooterEject extends CommandBase {
+  private ShooterSubsystem m_ShooterSubsystem;
+  private CDSSubsystem m_CDSSubsystem;
+  private int i;
+  private AimModes tempMode;
+
+  /** Creates a new ShooterPressed. */
+  public ShooterEject(ShooterSubsystem shooterSubsystem, CDSSubsystem cdsSubsystem) {
+    // Use addRequirements() here to declare subsystem dependencies.
+    addRequirements(shooterSubsystem);
+    addRequirements(cdsSubsystem);
+    m_ShooterSubsystem = shooterSubsystem;
+    m_CDSSubsystem = cdsSubsystem;
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    tempMode = m_ShooterSubsystem.getAimMode();
+    i = 0;
+    m_ShooterSubsystem.setAimMode(Constants.AimModes.EJECT);
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    m_ShooterSubsystem.prime();
+    if (m_ShooterSubsystem.wheelReady()) {
```

Comment:
if wheelready or i>0?  

If wheelready falls out of range, then we won't incrmement i during that loop